backend/app/core/rag/graph_builder.py

```
@@ -0,0 +1,130 @@
+        unique[key] = e
+    return list(unique.values())
+
+class KnowledgeGraphBuilder:
+    def __init__(self, cache_dir: str = None):
+        self.cache_dir = Path(cache_dir or settings.KB_CACHE_GRAPH_ROOT)
+        self.cache_dir.mkdir(parents=True, exist_ok=True)
+        self.morph = pymorphy3.MorphAnalyzer()
+
+    def _normalize_text(self, text: str) -> set:
+        #Лемматизация текста для нормализации ключевых слов
+        words = re.findall(r'\b[а-яА-ЯёЁa-zA-Z]{3,}\b', text.lower())
```

Ruff (0.15.12)

[warning] String contains ambiguous `а` (CYRILLIC SMALL LETTER A). Did you mean `a` (LATIN SMALL LETTER A)?

(RUF001)